Upgrade docs for 0.3.5 and 0.3.6

conorsch opened the pull request:
Adds new documentation for the upgrade procedure for releases `0.3.5` and `0.3.6`. Both releases require admin intervention to update local files on the Admin Workstation, then run the Ansible playbooks. These changes are now documented publicly (see #1173), and will be available on docs.securedrop.org after merging.

To review the docs offline prior to merging, run:

```
cd docs/
make html
```

Then open `_build/html/index.html` in your favorite browser and review the additions to the **Upgrade SecureDrop** section in the navbar. If you have trouble, see the [Documentation Guidelines](https://docs.securedrop.org/en/latest/development/documentation.html) (thanks, @garrettr!).

garrettr commented on docs/upgrade/0.3.5-to-0.3.6.rst:
```diff
@@ -0,0 +1,100 @@
+SecureDrop 0.3.6 was a maintenance release to update the expiration
+date on the GPG signing key used for the Freedom of the Press
+apt repository. The signing key expired on October 26, 2015, preventing
+automatically updates from Freedom of the Press until the issue is
```
"preventing automatically updates" is grammatically incorect

garrettr commented on docs/upgrade/0.3.5-to-0.3.6.rst:
```diff
@@ -0,0 +1,100 @@
+   (these commands are from the :doc:`../set_up_admin_tails` guide) ::
+
+    gpg --keyserver pool.sks-keyservers.net --recv-key B89A29DB2128160B8E4B1B4CBADDE0C7FC9F6818
+    gpg --fingerprint B89A29DB2128160B8E4B1B4CBADDE0C7FC9F6818
```
We should add a note, here or as a link from here to a dedicate document, about verifying this key. It is available on numerous HTTPS sites, and it is also signed by all of the developer's personal keys, so it should be possible to verify it via the GPG WoT as well.

conorsch replied:
Done.

garrettr replied:
Nice :+1:

garrettr commented:
Nit: I think hard wrapping the text is good (makes it easier to diff on Github, easier to read in certain editors, etc.), but you should pick a *consistent* line length if you do. It's all over the place in these docs. 79 characters is usually good IMO.

garrettr commented on docs/upgrade/0.3.x-to-0.3.5.rst:
```diff
@@ -0,0 +1,119 @@
+   .. note:: SecureDrop upgrades typically recommend using the simpler
+             ``git stash pop`` command to restore instance-specific settings.
+             Since 0.3.5 makes changes to the ``prod-specific.yml`` file,
+             more verbose git commands are necessary to avoid a merge conflict.
```
:+1: Good explanation!

garrettr commented on docs/upgrade/0.3.x-to-0.3.5.rst:
```diff
@@ -0,0 +1,119 @@
+    ansible-playbook -i inventory -s -u <user> securedrop-prod.yml
+
+During the playbook run, the Postfix TLS verification settings will be updated
+on the Monitor Server, the AppArmor profiles will be updated on the Application Server,
```
Nit: the AppArmor profiles are updated automatically with version 0.3.5 of the `securedrop-app-code` package, which happens independently of the Ansible playbook run for 0.3.5.

conorsch commented:
@garrettr Addressed all nits and rewrapped lines at 79 chars. (There's still one code line that's over length, and I didn't fight with it.) r?

ageis commented on docs/upgrade/0.3.5-to-0.3.6.rst:
```diff
@@ -0,0 +1,112 @@
+manually resolved.
+
+
+.. note:: We're considering creating a separate ``securedrop-keyring`` 
```
Did this during the hackathon, so maybe you could think about removing this note. I hadn't thought of naming it `securedrop-keyring`, I went with Tor's convention instead. Maybe `securedrop-keyring` is better.

https://github.com/freedomofpress/apt.freedom.press-keyring

conorsch replied:
@ageis Great catch, I've removed the specific name in d2212a2. Off the cuff I like the sound of `securedrop-keyring`, but we can discuss pros and cons in a separate issue.

ageis commented:
Looks good to me.

garrettr commented on docs/upgrade/0.3.5-to-0.3.6.rst:
```diff
@@ -0,0 +1,110 @@
+During the playbook run, the GPG signing key will be updated, and all packages
+will be upgraded on both servers. The new signing key is valid until October 
+2016,
+at which point another manual update may be necessary.
```
Line wrapping is weird here.

garrettr commented on docs/upgrade/0.3.5-to-0.3.6.rst:
```diff
@@ -0,0 +1,110 @@
+
+SecureDrop 0.3.6 was a maintenance release to update the expiration
+date on the GPG signing key used for the Freedom of the Press
+apt repository. The signing key expired on October 26, 2015, preventing
```
Rewrap this paragraph? This line is too short, for one.

garrettr commented on docs/upgrade/0.3.x-to-0.3.5.rst:
```diff
@@ -0,0 +1,124 @@
+SecureDrop 0.3.5 updates how TLS certificates for the SMTP relay
+used to send OSSEC notifications are handled. It also updates
+the AppArmor profile for Apache to ensure the service starts
+successfully.
```
Rewrap this too?

garrettr commented on docs/upgrade/0.3.x-to-0.3.5.rst:
```diff
@@ -0,0 +1,124 @@
+``prod-specific.yml`` file and run a special git command designed to merge
+site-specific changes in safely.
+
+
```
Nit: unnecessary newline.

garrettr commented on docs/upgrade/0.3.x-to-0.3.5.rst:
```diff
@@ -0,0 +1,124 @@
+
+#. Open ``prod-specific.yml`` in your preferred text editor.
+   Delete the line that starts with ``smtp_relay_fingerprint`` and save the 
+file.
```
Line wrapping is weird here.

garrettr commented:
Once my line comments are addressed, r+

conorsch commented:
Thank you for your careful review, @ageis @garrettr! Addressed all whitespace and wrapping concerns, as well as the redundant "package" to, and rebased into the final commit.

ageis commented:
@conorsch Sphinx is showing these minor issues:
```
/home/kevin/dev/securedrop/docs/upgrade/0.3.5-to-0.3.6.rst:22: WARNING: Title underline too short.

Update Expired GPG Signing Key (Admin Workstation)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/kevin/dev/securedrop/docs/upgrade/0.3.5-to-0.3.6.rst:28: WARNING: Title underline too short.

Update Expired GPG Signing Key (SecureDrop Servers)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/kevin/dev/securedrop/docs/upgrade/0.3.5-to-0.3.6.rst:28: WARNING: Title underline too short.

Update Expired GPG Signing Key (SecureDrop Servers)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
